### PR TITLE
feat(LVGL port): Support for RGB screen

### PR DESCRIPTION
**Pull request by espzav**
# ESP-BSP Pull Request checklist

- [x] Waiting for merge #320
- [x] Version of modified component bumped
- [x] CI passing

# Change description
1: esp_lv_port adds RGB interface support

All BSPs will be updated in next PR. 

Many thanks to @espressif2022 for very well support! This PR is made from #313

### CODE REVIEW

File: components/esp_lvgl_port/src/lvgl9/esp_lvgl_port_disp.c [hunk 1]
```diff
@@ -319,4 +456,20 @@ static void lvgl_port_disp_size_update_callback(lv_event_t *e)
...
+{
+    assert(e);
+    lvgl_port_display_ctx_t *disp_ctx = (lvgl_port_display_ctx_t *)e->user_data;
+    lvgl_port_disp_rotation_update(disp_ctx);
+}
+
+static void lvgl_port_display_invalidate_callback(lv_event_t *e)
+{
+    /* Wake LVGL task, if needed */
+    lvgl_port_task_wake(LVGL_PORT_EVENT_DISPLAY);
```
Comment (espressif2022) [inline components/esp_lvgl_port/src/lvgl9/esp_lvgl_port_disp.c:474]: https://github.com/lvgl/lvgl/blob/addc3f0c7327f0945d560f06f6cc2487a223a0ef/src/drivers/nuttx/lv_nuttx_lcd.c#L119C13-L119C23
	
In this cb, can we add a size readjustment function, similar to the implementation of the above link.
Due to some LCD rendering sizes must be 4-byte aligned.

Comment (espzav) [reply components/esp_lvgl_port/src/lvgl9/esp_lvgl_port_disp.c:474]: This is hard to add now. It is hard to test for me, if I haven't got specific HW. I can add it into my notes and add this feature later.

Comment (espressif2022) [reply components/esp_lvgl_port/src/lvgl9/esp_lvgl_port_disp.c:474]: > This is hard to add now. It is hard to test for me, if I haven't got specific HW. I can add it into my notes and add this feature later.

ok, we add it next step, and I'll prepare one pcs for you later.

File: components/esp_lvgl_port/src/lvgl9/esp_lvgl_port_disp.c [hunk 2]
```diff
@@ -277,14 +402,26 @@ static void lvgl_port_flush_callback(lv_display_t *drv, const lv_area_t *area, u
...
+    if (disp_ctx->disp_type == LVGL_PORT_DISP_TYPE_RGB && (disp_ctx->flags.full_refresh || disp_ctx->flags.direct_mode)) {
+        if (lv_disp_flush_is_last(drv)) {
+            /* If the interface is I80 or SPI, this step cannot be used for drawing. */
+            esp_lcd_panel_draw_bitmap(disp_ctx->panel_handle, offsetx1, offsety1, offsetx2 + 1, offsety2 + 1, color_map);
+            /* Waiting for the last frame buffer to complete transmission */
+            ulTaskNotifyValueClear(NULL, ULONG_MAX);
+            ulTaskNotifyTake(pdTRUE, portMAX_DELAY);
+        }
+    } else {
+        esp_lcd_panel_draw_bitmap(disp_ctx->panel_handle, offsetx1, offsety1, offsetx2 + 1, offsety2 + 1, color_map);
```
[the remark this answers is not in the document]
Comment (espressif2022) [reply components/esp_lvgl_port/src/lvgl9/esp_lvgl_port_disp.c:415]: Yes, the flag bit is better, so users don’t need to care about our IDF RGB driver, and they can reduce the init steps.

File: components/esp_lvgl_port/src/lvgl9/esp_lvgl_port_disp.c (hunk 1 above)
Comment (espzav) [reply components/esp_lvgl_port/src/lvgl9/esp_lvgl_port_disp.c:482]: Thank you!

**Comment by espzav** [comment]
One issue is here. When I am using display (https://github.com/espressif/esp-bsp/tree/master/examples/display) example with SPI/DSI display, it is not working properly with this line: https://github.com/espressif/esp-bsp/blob/81a36703ea8743553788c5e8a718642738a07de9/components/esp_lvgl_port/src/lvgl9/esp_lvgl_port.c#L230 and this line https://github.com/espressif/esp-bsp/blob/81a36703ea8743553788c5e8a718642738a07de9/components/esp_lvgl_port/src/lvgl9/esp_lvgl_port.c#L216 
I am working on it. I don't know, how to handle right, for maximum task sleep.

**Comment by espzav** [comment]
@tore-espressif PTAL

**Comment by espzav** [comment]
> One issue is here. When I am using display (https://github.com/espressif/esp-bsp/tree/master/examples/display) example with SPI/DSI display, it is not working properly with this line:
> 
> https://github.com/espressif/esp-bsp/blob/81a36703ea8743553788c5e8a718642738a07de9/components/esp_lvgl_port/src/lvgl9/esp_lvgl_port.c#L230
> 
> and this line
> https://github.com/espressif/esp-bsp/blob/81a36703ea8743553788c5e8a718642738a07de9/components/esp_lvgl_port/src/lvgl9/esp_lvgl_port.c#L216
> 
> 
> I am working on it. I don't know, how to handle right, for maximum task sleep.

Fixed on master.

**Comment by espzav** [comment]
@espressif2022 and @tore-espressif Do we want to update BSP for `esp32_s3_lcd_ev_board` in this PR or should we make new PR (or update #313)?

**Comment by espressif2022** [comment]
> @espressif2022 and @tore-espressif Do we want to update BSP for `esp32_s3_lcd_ev_board` in this PR or should we make new PR (or update #313)?

There is an RGB example in the test case. I think this PR does not need to be updated BSP of esp32_s3_lcd_ev_board.

File: components/esp_lvgl_port/examples/rgb_lcd/main/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+set(LV_DEMO_DIR ../managed_components/lvgl__lvgl/demos)
+file(GLOB_RECURSE LV_DEMOS_SOURCES ${LV_DEMO_DIR}/*.c)
```
Comment (tore-espressif) [inline components/esp_lvgl_port/examples/rgb_lcd/main/CMakeLists.txt:2]: This should no longer be needed. When you enable `CONFIG_LV_USE_DEMO_WIDGETS` in menuconfig, the demo source files will be added to the build

https://github.com/lvgl/lvgl/blob/master/env_support/cmake/esp.cmake#L21

File: components/esp_lvgl_port/examples/rgb_lcd/main/idf_component.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+dependencies:
+  idf: ">=4.4"
```
Comment (tore-espressif) [inline components/esp_lvgl_port/examples/rgb_lcd/main/idf_component.yml:2]:
```suggestion
dependencies:
  idf: ">=5.0"
```
5.0 for RGB support?

**Review by tore-espressif — COMMENTED** [review]
@espzav Thank you for the PR, I'm leaving a WIP review comments, just so you aren't blocked

**Review by tore-espressif — APPROVED** [review]

**Pull request merged**